satpy/tests/test_composites.py

```diff
@@ -1164,15 +1164,72 @@ def test_multiple_sensors(self):
         assert res.attrs['sensor'] == {'abi', 'glm'}
 
 
-class TestMaskingCompositor(unittest.TestCase):
+class TestMaskingCompositor:
     """Test case for the simple masking compositor."""
 
+    @pytest.fixture
+    def conditions_v1(self):
```

Comment:
I've never seen a pytest fixture as a class method. Does this provide any benefit? It seems unnecessary to have the `self` parameter.

Reply:
Did you mean as an instance method?

Having them inside the class can have the benefit of the class scope, but I'm not using that at the moment.  

I can make them static methods or take them out of the class entirely, if you prefer.

Reply:
The rest of the test methods don't use `self` either since my refactoring, so I could make them all static methods or remove the entire class (although in that case, codescene might complain about too many functions).

What do you think?

Reply:
I guess leave them as is. I've had some of the code quality checkers complain about methods like this that they should/could be static. If they aren't complaining here then this is fine to me.